api/controllers/FlairController.js

```diff
@@ -130,88 +129,93 @@ module.exports = {
-          similar_banned_fcs.join(', ') + '\n\n';
-      }
-      if (banned_alts.length) {
-        message += 'This user may be an alt of the banned user' + (banned_alts.length === 1 ? '' : 's') + ' /u/' + banned_alts.join(', /u/') + '\n\n';
-      }
-      if (flagged.length) {
-        message += 'The friend code' + (flagged.length === 1 ? ' ' + flagged + ' is' : 's ' + flagged.join(', ') + ' are') + ' invalid.\n\n';
-        var formattedNote = "Invalid friend code" + (flagged.length == 1 ? "" : "s") + ": " + flagged.join(', ');
-        promises.push(Usernotes.addUsernote(refreshToken, 'FlairHQ', 'pokemontrades', req.user.name, formattedNote, 'spamwarn', ''));
+      if (!blockReport && (identical_banned_fcs.length || similar_banned_fcs.length || banned_alts.length || flagged.length)) {
+        var message = 'The user /u/' + req.user.name + ' set the following flairs:\n\n' + flairs.ptrades + '\n\n' + flairs.svex + '\n\n';
+        if (identical_banned_fcs.length) {
+          message += 'This flair contains a banned friend code: ' + identical_banned_fcs + '\n\n';
+        } else if (flagged.length && similar_banned_fcs.length) {
+          message += 'This flair contains a friend code similar to the following banned friend code'  + (similar_banned_fcs.length > 1 ? 's: ' : ': ') +
+            similar_banned_fcs.join(', ') + '\n\n';
+        }
+        if (banned_alts.length) {
+          message += 'This user may be an alt of the banned user' + (banned_alts.length === 1 ? '' : 's') + ' /u/' + banned_alts.join(', /u/') + '\n\n';
+        }
+        if (flagged.length) {
+          message += 'The friend code' + (flagged.length === 1 ? ' ' + flagged + ' is' : 's ' + flagged.join(', ') + ' are') + ' invalid.\n\n';
+          var formattedNote = "Invalid friend code" + (flagged.length == 1 ? "" : "s") + ": " + flagged.join(', ');
+          promises.push(Usernotes.addUsernote(refreshToken, 'FlairHQ', 'pokemontrades', req.user.name, formattedNote, 'spamwarn', ''));
+        }
+        message = message.slice(0,-2);
+        promises.push(Reddit.sendPrivateMessage(refreshToken, "FlairHQ notification", message, "/r/pokemontrades"));
       }
-      message = message.slice(0,-2);
-      promises.push(Reddit.sendPrivateMessage(refreshToken, "FlairHQ notification", message, "/r/pokemontrades"));
+      Promise.all(promises).then(function () {
+        Event.create([{
+          type: "flairTextChange",
+          user: req.user.name,
+          content: "Changed PokemonTrades flair text to: " + req.allParams().ptrades + ". IP: " + ipAddress
+        }, {
+          type: "flairTextChange",
+          user: req.user.name,
+          content: "Changed SVExchange flair text to: " + req.allParams().svex + ". IP: " + ipAddress
+        }]).exec(function () {});
+        return res.ok(req.user);
+      });
+    } catch (err) {
+      return res.serverError(err);
```

Comment:
This is all the same as before, just dumped it into a try/catch since it's async.